### PR TITLE
 PLAT-150-Provide_support_for_Gov_cloud

### PR DESCRIPTION
**Pull request by MenahemAboush**

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (itzikYeret) [inline .idea/.gitignore]: please remove .idea directory

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,10 @@ module github.com/terraform-providers/terraform-provider-dome9
 
 go 1.13
 
+replace (
+	github.com/dome9/dome9-sdk-go => ../dome9-sdk-go
```
Comment (itzikYeret) [inline go.mod]: remove after sdk tag publish

File: .idea/.gitignore (hunk 1 above)
Comment (itzikYeret) [reply .idea/.gitignore]: also remove the exe file

File: website/docs/r/cloudaccount_aws.html.markdown [hunk 3]
```diff
@@ -111,6 +111,31 @@ resource "dome9_cloudaccount_AWS" "test" {
 }
 ```
 
+```hcl
+resource "dome9_cloudaccount_AWSGOV" "test" {
+  name  = "ACCOUNT NAME"
+ 
+  credentials  {
```
Comment (itzikYeret) [inline website/docs/r/cloudaccount_aws.html.markdown:122]: if you set aws gov you should use UserBased

File: website/docs/r/cloudaccount_aws.html.markdown [hunk 4]
```diff
@@ -111,6 +111,31 @@ resource "dome9_cloudaccount_AWS" "test" {
 }
 ```
 
+```hcl
+resource "dome9_cloudaccount_AWSGOV" "test" {
```
Comment (itzikYeret) [inline website/docs/r/cloudaccount_aws.html.markdown]: there is no resource dome9_cloudaccount_AWSGOV but dome9_cloudaccount_aws with vendor awsgov

File: website/docs/r/cloudaccount_aws.html.markdown [hunk 5]
```diff
@@ -122,18 +147,18 @@ The following arguments are supported:
...
 
 `credentials` has the following arguments:
-
-* `arn` - (Required) AWS Role ARN (to be assumed by Dome9)
-* `secret` - (Required) The AWS role External ID (Dome9  will have to use this secret in order to assume the role)
-* `type` - (Required) The cloud account onboarding method. Set to "RoleBased".
+*  `arn`       - (Optional) AWS Role ARN (to be assumed by Dome9 - Required for AWS)
+*  `secret`    - (Required) The AWS role External ID (Dome9  will have to use this secret in order to assume the role)
+*  `type`      - (Required) The cloud account onboarding method. Set to "RoleBased".
+*  `api_key`   - (Optional) AWS Role ARN (to be assumed by Dome9 - Required for AWSGOV)
```
Comment (itzikYeret) [inline website/docs/r/cloudaccount_aws.html.markdown]: this is not Role arn but user api key

File: website/docs/r/cloudaccount_aws.html.markdown [hunk 6]
```diff
@@ -122,18 +147,18 @@ The following arguments are supported:
 ### Credentials
 
 `credentials` has the following arguments:
-
-* `arn` - (Required) AWS Role ARN (to be assumed by Dome9)
-* `secret` - (Required) The AWS role External ID (Dome9  will have to use this secret in order to assume the role)
-* `type` - (Required) The cloud account onboarding method. Set to "RoleBased".
+*  `arn`       - (Optional) AWS Role ARN (to be assumed by Dome9 - Required for AWS)
+*  `secret`    - (Required) The AWS role External ID (Dome9  will have to use this secret in order to assume the role)
+*  `type`      - (Required) The cloud account onboarding method. Set to "RoleBased".
```
Comment (itzikYeret) [inline website/docs/r/cloudaccount_aws.html.markdown]: or user based

File: website/docs/r/cloudaccount_aws.html.markdown [hunk 7]
```diff
@@ -122,18 +147,18 @@ The following arguments are supported:
 ### Credentials
 
 `credentials` has the following arguments:
-
-* `arn` - (Required) AWS Role ARN (to be assumed by Dome9)
-* `secret` - (Required) The AWS role External ID (Dome9  will have to use this secret in order to assume the role)
-* `type` - (Required) The cloud account onboarding method. Set to "RoleBased".
+*  `arn`       - (Optional) AWS Role ARN (to be assumed by Dome9 - Required for AWS)
+*  `secret`    - (Required) The AWS role External ID (Dome9  will have to use this secret in order to assume the role)
```
Comment (itzikYeret) [inline website/docs/r/cloudaccount_aws.html.markdown]: this is not only role externalId, if we use apikey, this is the secret

**Review by itzikYeret — CHANGES_REQUESTED** [review]

File: dome9/resource_dome9_cloudaccount_aws.go [hunk 8]
```diff
@@ -306,20 +303,33 @@ func resourceCloudAccountAWSUpdate(d *schema.ResourceData, meta interface{}) err
 }
 
 func expandCloudAccountAWSRequest(d *schema.ResourceData) aws.CloudAccountRequest {
+	credentials := expandCloudAccountAWSCredentials(d)
+	vendor := "aws"
```
Comment (itzikYeret) [inline dome9/resource_dome9_cloudaccount_aws.go]: why is the vendor not taken from the resource? where is the vendor validation against the credentials?

File: website/docs/r/cloudaccount_aws.html.markdown [hunk 9]
```diff
@@ -111,6 +111,31 @@ resource "dome9_cloudaccount_AWS" "test" {
 }
 ```
 
+```hcl
+resource "dome9_cloudaccount_awsgov" "test" {
```
Comment (itzikYeret) [inline website/docs/r/cloudaccount_aws.html.markdown]: there is no resource dome9_cloudaccount_awsgov. remove the "gov"

File: website/docs/r/cloudaccount_aws.html.markdown [hunk 10]
```diff
@@ -111,6 +111,31 @@ resource "dome9_cloudaccount_AWS" "test" {
...
+  name  = "ACCOUNT NAME"
+
```
Comment (itzikYeret) [inline website/docs/r/cloudaccount_aws.html.markdown]: add vendor property to awsgov

File: website/docs/r/cloudaccount_aws.html.markdown [hunk 11]
```diff
@@ -122,23 +147,23 @@ The following arguments are supported:
 ### Credentials
 
 `credentials` has the following arguments:
-
-* `arn` - (Required) AWS Role ARN (to be assumed by Dome9)
-* `secret` - (Required) The AWS role External ID (Dome9  will have to use this secret in order to assume the role)
-* `type` - (Required) The cloud account onboarding method. Set to "RoleBased".
+*  `arn`       - (Optional) AWS Role ARN (to be assumed by Dome9 - Required for AWS but not for awsGov)
+*  `secret`    - (Required) The AWS role External ID or AWS user secret key (Dome9  will have to use this secret)
```
Comment (itzikYeret) [inline website/docs/r/cloudaccount_aws.html.markdown]: role externalid for rolebased and user secret for userbased

**Review by itzikYeret — CHANGES_REQUESTED** [review]

File: website/docs/r/cloudaccount_aws.html.markdown (hunk 10 above)
Comment (MenahemAboush) [reply website/docs/r/cloudaccount_aws.html.markdown:117]: U right... it's always dome9_cloudaccount_aws.
my bad.

File: dome9/resource_dome9_cloudaccount_aws.go [hunk 12]
```diff
@@ -305,22 +304,109 @@ func resourceCloudAccountAWSUpdate(d *schema.ResourceData, meta interface{}) err
...
+		if err != nil {
+			return err
+		}
+		break
+	}
+	return nil
+}
+
+func validateAwsGovVendor(credentials aws.CloudAccountCredentials, regions []string) (bool, error) {
+	awsGovRegions := map[string]bool{"us_gov_east_1": true, "us_gov_west_1": true}
```
Comment (itzikYeret) [inline dome9/resource_dome9_cloudaccount_aws.go]: you don't need this anymore, use only slice and not map

File: dome9/resource_dome9_cloudaccount_aws.go [hunk 13]
```diff
@@ -305,22 +304,109 @@ func resourceCloudAccountAWSUpdate(d *schema.ResourceData, meta interface{}) err
...
+	}
+
+	if credentials.Type != "UserBased" || len(credentials.ApiKey) == 0 || len(credentials.Secret) == 0 {
+		return false, fmt.Errorf("awsGov vendor has wrong credentials")
+	}
+	return true, nil
+}
+
+func validateAwsVendor(credentials aws.CloudAccountCredentials, regions []string) (bool, error) {
+	awsRegions := map[string]bool{"us_east_1": true, "us_west_1": true, "eu_west_1": true, "ap_southeast_1": true, "ap_northeast_1": true, "us_west_2": true,
```
Comment (itzikYeret) [inline dome9/resource_dome9_cloudaccount_aws.go]: you don't need this anymore, use only slice and not map

File: dome9/resource_dome9_cloudaccount_aws_test.go [hunk 14]
```diff
@@ -142,6 +142,26 @@ data "%s" "%s" {
 	)
 }
 
+func testAccCheckCloudAccountAWSGOVBasic(resourceTypeAndName, generatedName, resourceName, arn, additionalBlock string) string {
```
Comment (itzikYeret) [inline dome9/resource_dome9_cloudaccount_aws_test.go]: remove

**Review by itzikYeret — CHANGES_REQUESTED** [review]

**Review by itzikYeret — APPROVED** [review]

**Pull request merged**